crates/anvil/src/evm.rs

```diff
@@ -0,0 +1,91 @@
+use std::{fmt::Debug, sync::Arc};
+
+use alloy_primitives::Address;
+use foundry_evm::revm::{self, precompile::Precompile, ContextPrecompile, ContextPrecompiles};
+
+/// Object-safe trait that enables injecting extra precompiles when using
+/// `anvil` as a library.
+pub trait PrecompileFactory: Send + Sync + Unpin + Debug {
+    /// Returns a set of precompiles to extend the EVM with.
+    fn precompiles(&self) -> Vec<(Address, Precompile)>;
+}
+
+/// Appends a handler register to `evm` that injects the given `precompiles`.
+pub fn inject_precompiles<DB, I>(
+    evm: &mut revm::Evm<'_, I, DB>,
+    precompiles: Vec<(Address, Precompile)>,
+) where
+    DB: revm::Database,
+    I: revm::Inspector<DB>,
```

Comment:
I think we don't even need to restrict this type?

Reply:
Yeah, you are right, copy/paste error